rust/lazy/src/lazy_pathlist_config.rs

```diff
@@ -0,0 +1,285 @@
+use std::{
+    collections::HashSet,
+    fs::File,
+    io::{prelude::*, BufRead, BufReader},
+    ops::{Deref, DerefMut},
+    path::{Path, PathBuf},
+};
+
+use tracing::{error, info};
+
+use crate::{error::Result, lazy_rule::LazyStrategy};
+
+#[derive(Debug)]
+pub struct LazyPathListConfig {
```

Comment:
Small thing, but could save time on larger repos -- we could do an optimization here by saving one large String that is the entire loaded file, then we store a HashSet of &str objects with the same lifetime (and could also then count newlines and use HashSet::with_capacity).

Reply:
How about file names with newlines?

Reply:
This seems like a poor micro-optimization with no clear benefits.